lib/utils/errors.js

```diff
@@ -35,6 +63,55 @@ class ExceededMaxFileSizeError extends EvervaultError {}
 
 class TokenCreationError extends EvervaultError {}
 
+const mapFunctionFailureResponseToError = ({ error, id }) => {
+  if (error) {
+    if (error.message.includes('The function failed to initialize.')) {
+      throw new FunctionInitializationError(error.message, error.stack, id);
+    }
+    throw new FunctionRuntimeError(error.message, error.stack, id);
+  }
+  throw new EvervaultError('An unknown error occurred.');
+};
+
+const mapFunctionResponseToError = ({ code, detail }) => {
+  if (code === 'unauthorized') {
+    throw new UnauthorizedError(detail);
+  }
+  if (code === 'forbidden') {
+    if (detail == 'Run requested by an IP address which is not whitelisted.') {
+      throw new ForbiddenIPError(detail);
+    }
+    throw new ForbiddenError(detail);
+  }
+  if (code === 'resource-not-found') {
+    throw new FunctionNotFoundError(detail);
+  }
+  if (code === 'request-timeout') {
+    throw new FunctionTimeoutError(detail);
+  }
+  if (code === 'function-not-ready') {
+    throw new FunctionNotReadyError(detail);
+  }
+  if (code === 'invalid-request') {
+    if (
+      detail ==
+        'The Function name in token does not match the Function name in the request.' ||
+      detail ==
+        'The Payload in the request does not match the payload used to create the token'
+    ) {
+      throw new FunctionRunTokenMismatchError(detail);
+    }
+    throw new RequestError(detail);
+  }
+  if (code === 'method-not-allowed') {
```

Comment:
I believe this should actually never been thrown as this would be caused by an implementation error in the SDK.

Reply:
I agree, I included it because I conflated `EvervaultError` with `500` in my head and this wasn't server side, but `EvervaultError` is a bit of a catchall here so it's actually fine. Will update 👍